src/synthesizer.ts

```diff
@@ -417,8 +472,107 @@ export class PulumiSynthesizer extends PulumiSynthesizerBase implements cdk.IReu
         return this.cloudFormationLocationFromFileAsset(location);
     }
 
+    /**
+     * Gets registry credentials for the given ECR repository
+     *
+     * @param repo - The ECR repository to get the credentials for
+     * @returns The registry credentials for the ECR repository
+     */
+    private getEcrCredentialsOutput(repo: aws.ecr.Repository): docker.types.input.RegistryArgs {
+        const ecrCredentials = aws.ecr.getCredentialsOutput(
+            {
+                registryId: repo.registryId,
+            },
+            { parent: this.stagingStack },
+        );
+        return ecrCredentials.authorizationToken.apply((token) => {
+            const decodedCredentials = Buffer.from(token, 'base64').toString();
+            const [username, password] = decodedCredentials.split(':');
+            if (!password || !username) {
+                throw new Error('Invalid credentials');
+            }
+            return {
+                address: ecrCredentials.proxyEndpoint,
+                username: username,
+                password: password,
+            };
+        });
+    }
+
     addDockerImageAsset(asset: cdk.DockerImageAssetSource): cdk.DockerImageAssetLocation {
-        throw new Error('Docker image assets are not supported yet');
+        assertBound(this.outdir);
+        if (asset.executable || !asset.directoryName) {
+            throw new Error(`Docker image assets produced by commands are not yet supported`);
+        }
+
+        const { repoName, repo } = this.getCreateRepo(asset);
+        const imageTag = asset.sourceHash;
+        const canonicalImageName = pulumi.interpolate`${repo.repositoryUrl}:${imageTag}`;
+        const context = path.join(this.outdir, asset.directoryName);
+        const dockerFile = path.join(context, asset.dockerFile ?? 'Dockerfile');
+        const assetKey = `${asset.assetName}:${asset.sourceHash}`;
+
+        // Assets can be registered multiple times, but we should only create the resource once
+        if (this.seenImageAssets.has(assetKey)) {
+            return {
+                repositoryName: repoName,
+                imageUri: asString(this.seenImageAssets.get(assetKey)!.ref),
+            };
+        }
+
+        const registryCredentials = this.getEcrCredentialsOutput(repo);
+
+        const cacheFrom = fromCdkCacheFrom(asset.dockerCacheFrom);
+        const cacheTo = fromCdkCacheTo(asset.dockerCacheTo);
+
+        const image = new docker.Image(
```

Comment:
Not sure if there are any defaults we should be setting for any of these settings (e.g. cache stuff)

Reply:
Does CDK have any defaults here?

Reply:
No CDK doesn't use any defaults

https://github.com/cdklabs/cdk-assets/blob/e1da0e1a86ab9af3880368c65a988dd5db1e95ea/lib/private/docker.ts#L96-L128